src/main/java/model/input/file/DataFile.java

```diff
@@ -311,35 +261,104 @@ public void createMetaDataValue(String name, String type) {
 		try {
 			String fileName = this.getFile().getName();
 			String metaValue = fileName.substring(0, fileName.lastIndexOf("."));
-
-			this.metaDataValue = parseDataValue(metaValue, type);
+			Class<? extends DataValue> typeClass = DataFile.getColumnType(type);
+			this.metaDataValue = parseSimpleDataValue(metaValue, typeClass);
 			this.setMetaDataType(getColumnType(type));
 			this.setMetaDataColumnName(name);
+			hasMetaData = true;
 		} catch (FileNotFoundException e) {
 			log.log(Level.SEVERE, "The file could not be found", e);
 		}
 	}
 
 	/**
-	 * Parses a string to a DataValue of a given type.
+	 * Parses a String or number to a DataValue of a given type.
 	 * @param value The value to parse
 	 * @param type The type of DataValue to parse to
 	 * @return The new created DataValue
 	 */
-	public static DataValue parseDataValue(String value, String type) {
-		switch (type) {
-			case "int":
-				return new IntValue(Integer.parseInt(value));
-			case "float":
-				return new FloatValue(Float.parseFloat(value));
-			case "string":
-				return new StringValue(value);
-			default:
-				throw new RuntimeException("Class has not yet implemented");
+	protected DataValue parseSimpleDataValue(String value, Class<? extends DataValue> type) {
```

Comment:
Het zou waarschijnlijk mooier zijn als je deze methodes verplaatst naar ColumnInfo, maar dat is werk en nu hebben we niet echt tijd.